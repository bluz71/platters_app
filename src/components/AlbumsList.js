import React from 'react';
import PropTypes from 'prop-types';
import pluralize from 'pluralize';
import { Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import '../styles/AlbumsList.css';

const AlbumsList = ({ albums, onYear, onGenre }) => {
  return (
    albums.map(album =>
      <Col key={album.id} md={6}>
        <div className="Album">
          <h2>
            <Link to={`/artist/${album.artist_slug}/album/${album.album_slug}`}>{album.title}</Link>
          </h2>
          <h3>
            by <Link to={`/artist/${album.artist_slug}`}>{album.artist}</Link> <small>({album.tracks_count} {pluralize('Track', album.tracks_count)})</small>
          </h3>
          <div className="icon">
            <a onClick={() => onYear(album.year)}><FontAwesome name="calendar" /> {album.year}</a>
            <a onClick={() => onGenre(album.genre)}><FontAwesome name="tag" className="spacer-left-xs" /> {album.genre}</a>
            <FontAwesome name="comment-o" className="spacer-left-xs" /> {album.comments_count}
          </div>
          <Link to={`/artist/${album.artist_slug}/album/${album.album_slug}`}>
            <img className="img-responsive" src={album.cover_url} alt={album.title} />
          </Link>
          <ul>
            {album.tracks.map((track, index) => <li key={index}>{track}</li>)}
          </ul>
        </div>
      </Col>
    )
  );
};

AlbumsList.propTypes = {
  albums: PropTypes.array,
  onYear: PropTypes.func,
  onGenre: PropTypes.func
};

export default AlbumsList;
