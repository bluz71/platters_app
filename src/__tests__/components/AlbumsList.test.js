import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import AlbumsList from '../../components/AlbumsList';

const albums = () => {
  return [
    {
      'id': 1,
      'title': 'ABC',
      'artist': 'ABC',
      'artist_slug': 'abc',
      'album_slug': 'abc',
      'cover_url': 'http://localhost:3000/small_ABC--ABC.jpg',
      'tracks': [
        '1. First',
        '2. Second'
      ]
    },
    {
      'id': 2,
      'title': 'DEF',
      'artist': 'DEF',
      'artist_slug': 'def',
      'album_slug': 'def',
      'cover_url': 'http://localhost:3000/small_DEF--DEF.jpg',
      'tracks': [
        '1. First',
        '2. Second',
        '3. Third'
      ]
    }
  ];
};

describe('<AlbumsList />', () => {
  it('renders without crashing', () => {
    shallow(<AlbumsList albums={[]} />);
  });

  it('renders a list of albums', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
        <AlbumsList albums={albums()} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
