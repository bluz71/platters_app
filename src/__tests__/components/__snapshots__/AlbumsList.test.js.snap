// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumsList /> renders a list of albums 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AlbumsList
      albums={
        Array [
          Object {
            "album_slug": "abc",
            "artist": "ABC",
            "artist_slug": "abc",
            "cover_url": "http://localhost:3000/small_ABC--ABC.jpg",
            "id": 1,
            "title": "ABC",
            "tracks": Array [
              "1. First",
              "2. Second",
            ],
          },
          Object {
            "album_slug": "def",
            "artist": "DEF",
            "artist_slug": "def",
            "cover_url": "http://localhost:3000/small_DEF--DEF.jpg",
            "id": 2,
            "title": "DEF",
            "tracks": Array [
              "1. First",
              "2. Second",
              "3. Third",
            ],
          },
        ]
      }
    >
      <div
        className="AlbumsList"
      >
        <Col
          bsClass="col"
          componentClass="div"
          key="1"
          md={6}
        >
          <div
            className="col-md-6"
          >
            <div
              className="Album"
            >
              <h2>
                <Link
                  replace={false}
                  to="/artist/abc/album/abc"
                >
                  <a
                    href="/artist/abc/album/abc"
                    onClick={[Function]}
                  >
                    ABC
                  </a>
                </Link>
              </h2>
              <h3>
                by 
                <Link
                  replace={false}
                  to="/artist/abc"
                >
                  <a
                    href="/artist/abc"
                    onClick={[Function]}
                  >
                    ABC
                  </a>
                </Link>
                 
                <small>
                  (
                   
                  Tracks
                  )
                </small>
              </h3>
              <div
                className="icon"
              >
                <a
                  onClick={[Function]}
                >
                  <FontAwesome
                    name="calendar"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-calendar"
                    />
                  </FontAwesome>
                   
                </a>
                <a
                  onClick={[Function]}
                >
                  <FontAwesome
                    className="spacer-left-xs"
                    name="tag"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-tag spacer-left-xs"
                    />
                  </FontAwesome>
                   
                </a>
                <Link
                  replace={false}
                  to={
                    Object {
                      "pathname": "/artist/abc/album/abc",
                      "state": Object {
                        "scrollToComments": true,
                      },
                    }
                  }
                >
                  <a
                    href="/artist/abc/album/abc"
                    onClick={[Function]}
                  >
                    <FontAwesome
                      className="spacer-left-xs"
                      name="comment-o"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-comment-o spacer-left-xs"
                      />
                    </FontAwesome>
                     
                  </a>
                </Link>
              </div>
              <Link
                replace={false}
                to="/artist/abc/album/abc"
              >
                <a
                  href="/artist/abc/album/abc"
                  onClick={[Function]}
                >
                  <img
                    alt="ABC"
                    className="img-responsive"
                    src="http://localhost:3000/small_ABC--ABC.jpg"
                  />
                </a>
              </Link>
              <ul>
                <li
                  key="0"
                >
                  1. First
                </li>
                <li
                  key="1"
                >
                  2. Second
                </li>
              </ul>
            </div>
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="2"
          md={6}
        >
          <div
            className="col-md-6"
          >
            <div
              className="Album"
            >
              <h2>
                <Link
                  replace={false}
                  to="/artist/def/album/def"
                >
                  <a
                    href="/artist/def/album/def"
                    onClick={[Function]}
                  >
                    DEF
                  </a>
                </Link>
              </h2>
              <h3>
                by 
                <Link
                  replace={false}
                  to="/artist/def"
                >
                  <a
                    href="/artist/def"
                    onClick={[Function]}
                  >
                    DEF
                  </a>
                </Link>
                 
                <small>
                  (
                   
                  Tracks
                  )
                </small>
              </h3>
              <div
                className="icon"
              >
                <a
                  onClick={[Function]}
                >
                  <FontAwesome
                    name="calendar"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-calendar"
                    />
                  </FontAwesome>
                   
                </a>
                <a
                  onClick={[Function]}
                >
                  <FontAwesome
                    className="spacer-left-xs"
                    name="tag"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-tag spacer-left-xs"
                    />
                  </FontAwesome>
                   
                </a>
                <Link
                  replace={false}
                  to={
                    Object {
                      "pathname": "/artist/def/album/def",
                      "state": Object {
                        "scrollToComments": true,
                      },
                    }
                  }
                >
                  <a
                    href="/artist/def/album/def"
                    onClick={[Function]}
                  >
                    <FontAwesome
                      className="spacer-left-xs"
                      name="comment-o"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-comment-o spacer-left-xs"
                      />
                    </FontAwesome>
                     
                  </a>
                </Link>
              </div>
              <Link
                replace={false}
                to="/artist/def/album/def"
              >
                <a
                  href="/artist/def/album/def"
                  onClick={[Function]}
                >
                  <img
                    alt="DEF"
                    className="img-responsive"
                    src="http://localhost:3000/small_DEF--DEF.jpg"
                  />
                </a>
              </Link>
              <ul>
                <li
                  key="0"
                >
                  1. First
                </li>
                <li
                  key="1"
                >
                  2. Second
                </li>
                <li
                  key="2"
                >
                  3. Third
                </li>
              </ul>
            </div>
          </div>
        </Col>
      </div>
    </AlbumsList>
  </Router>
</MemoryRouter>
`;
